### DISCUSSION TITLE
Output as JSON

### DISCUSSION THREAD

**Discussion by rajyraman** [microsoft/powerplatform-vscode#323] [Ideas]
I was watching a video about [PowerShell Crescendo](https://www.youtube.com/watch?v=QDGgLUhwj8c&t=3387s) today. It seems like a great tool to wrap native commands, but in PPAC there is no choice of output format. For commands that return a table, it would be good to add an option to output as json as well. i.e pac application list, pac solution list, pac admin list, pac admin list-backups.

I was thinking of using PowerShell Crescendo to export all the unmanaged solutions in the environment in a single command using pac solution clone or pac solution sync. While I may be able to do it by parsing the table, it would be easier if there was a json option as well for every command that outputs a table.

Please let me know your thoughts.

**Comment by petrochuk** [comment]
These commands are working: "pac admin list --json", "pac connector list --json", and "pac org who --json"

**Comment by rajyraman** [comment]
@petrochuk - I didn't know about the --json param as it did not show up in `pac solution list help`. This makes it a lot easier. Thank you Petro. I have to find out all the commands that support this --json param.

**Reply by rajyraman** [reply]
@petrochuk -  `pac org fetch` does not support --json option. Can this please be standardised? IMHO, anything that shows a table should also have a --json option so that people can process it further using PowerShell if they want to.

For your consideration @kkanakas 🙏

**Reply by petrochuk** [reply]
This can be done without PAC CLI by simply calling WebAPI : [Use FetchXML with Web API](https://learn.microsoft.com/en-us/power-apps/developer/data-platform/webapi/use-fetchxml-web-api)

**Comment by petrochuk** [comment]
> @petrochuk - `pac org fetch` does not support --json option. Can this please be standardised? IMHO, anything that shows a table should also have a --json option so that people can process it further using PowerShell if they want to.
> 
> For your consideration @kkanakas 🙏

**Reply by rajyraman** [reply]
@petrochuk - Cross posting https://github.com/microsoft/Power-Fx/discussions/2734. Please consider adding

1. New --expression for a short Power Fx expression
2. --json flag to get the results after the Power Fx expression results are returned

cc @snizar007
